### DISCUSSION TITLE
Confused about SSR and Static Generation

### DISCUSSION THREAD

**Discussion by serena-ai-architect** [vercel/next.js#11213] [Help]
Recently I need to make a static official website with nextjs，and I am asked to do with SSR for a better SEO.    But nextjs tell us that Static Generation  pre-rendering is faster than SSR .   So how should i choose now ?  Anybody could help me to explain about these concepts ?  Thanks so much !

My Product Requirements：  a static official website,   SSR,  better SEO 

![image](https://user-images.githubusercontent.com/29164467/77134449-fde2ce80-6aa1-11ea-90ad-91de6905b629.png)

**Comment by timneutkens** [comment]
In your case the the "SSR" requirement is not really a requirement for server-rendering on-demand. They're asking for pre-rendering so that the HTML is sent to the user, both static generation and server-side rendering give this result.

You can read about pre-rendering here: https://nextjs.org/docs/basic-features/pages#pre-rendering

**Reply by serena-ai-architect** [reply]
Got it! Thanks a lot, Mr. Tim~   😋😉

**Comment by zimv** [comment]
A complete Sever-side Rendering process is: client request -> server route -> server component rendering -> server back。
Static Generation render component at build time. So a complete process is: client request -> server route -> server back static resources。

**Reply by serena-ai-architect** [reply]
感谢哈~
